python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -459,6 +480,9 @@ async def do_applyCorrection(self, id_data):
                                 self.model.get_correction_m2(azimuth,
                                                              elevation))
 
+        self.log.debug("Apply corrections from spectrograph")
+        await self.set_atspectrograph_corrections(azimuth, elevation)
```

Comment:
If you let `set_hexapod` take care of applying the hexapod correction, you could call this one first (to add its correction), then call `set_hexapod`.

Reply:
see comment in conversation about this suggestion/question